### ISSUE TITLE
Update or Remove  duplicate agent

### ISSUE THREAD

**Issue by cmiscloni** [wh1te909/tacticalrmm#685]
By reinstalling the agent, a duplicate computer appears on console, add the possibility to update or remove it in settings.
Thanks

**Comment by silversword411** [comment]
I noticed in mesh options there's a feature that "removes disconnected agents"

You might be able to set that to clean agents after x time not connected, but that might kill valid agent's

**Comment by cmiscloni** [comment]
The problem is when we 're reinstalling a computer (same computername) , a second agent is create. Maybe if it's possible to create a backgroung task which find the duplicate name and keep only the most recent ?

**Comment by dinger1986** [comment]
Is this for mesh or tactical?

**Comment by cmiscloni** [comment]
Tactical

**Comment by nr-plaxon** [comment]
Something along this maybe: #708

**Comment by bbrendon** [comment]
Sounds great but how on earth would you know its a duplicate?

**Comment by silversword411** [comment]
By MAC address, then hostname. Isn't that what everyone else does? ;)

**Comment by nr-plaxon** [comment]
MAC address by itself would work. Who puts a physical NIC to a different host nowadays… Maybe limit this to agents marked as clients and skip servers. Showing a list of duplicates in the process of the maintenance task so the user can cancel would also be a safety net.

**Comment by cmiscloni** [comment]
by IdentifyingNumber (serial)

**Comment by silversword411** [comment]
> by IdentifyingNumber (serial)

No, because there's machines with "to be filled in by OEM" out there

**Comment by silversword411** [comment]
You now have a manual method of bulk deleting old agents:
https://github.com/wh1te909/tacticalrmm/commit/2c8f207454884fa47ce245e4c3a6567e44cbd47b
https://github.com/wh1te909/tacticalrmm/pull/830/commits/3f3ab088d233734db0375681dd66fba2ed518154

By: agentver or days since last checkin

Will be in next release.

**Comment by jmoran-shafer** [comment]
+1
